packages/cli/src/cog/types.ts

```diff
@@ -2,78 +2,118 @@ import { BoundingBox, EpsgCode } from '@basemaps/geo';
 import { GdalCogBuilderResampling } from '../gdal/gdal.config';
 
-export interface CogJob {
-    /** Unique processing Id */
-    id: string;
+export interface CogSourceProperties {
+    files: NamedBounds[];
+    'proj:epsg': EpsgCode;
+    /** Ground Sampling Distance in meters per pixel of the generated imagery */
+    'eo:gsd': number;
+}
+
+export interface CogOutputProperties {
+    /** The bounds of all the cogs */
+    bounds: BoundingBox;
+    /** list of files to generate and their bounds */
+    files: NamedBounds[];
+    resampling: GdalCogBuilderResampling;
+    nodata?: number;
+    /**
+     * Quality level to use
+     * @default 90
+     */
+    quality: number;
+    /** Cutline options */
+    cutline?: {
+        href: string;
+        blend: number;
+    };
 
+    /** Vrts will add a second alpha layer if one exists, so dont always add one */
+    addAlpha: boolean;
+}
+
+export interface CogGenerated {
+    /** ISO date string */
+    datetime: string;
+    /** Package name of the generator */
+    package: string;
+    /* Version of the generator */
+    version: string;
+    /** Commit hash of the generator */
+    hash: string | undefined;
+}
+
+export interface CogBuild {
     /** Imagery set name */
     name: string;
+    /** source imagery properties */
+    source: CogSourceProperties;
+    sourceLocation: FileConfig;
 
-    /** Output projection */
-    projection: EpsgCode;
+    output: CogOutputProperties;
+    outputLocation: FileConfig;
+    generated: CogGenerated;
+}
 
-    source: {
-        /** List of input files */
-        files: NamedBounds[];
-
-        /** The number of pixels per meter for the best source image */
-        pixelScale: number;
-        /**
-         * The zoom level that corresponds approximately what the pixelScale of the source is
-         * for high quality aerial imagery this is generally 20-22 in Google Mercator
-         */
-        resZoom: number;
-        /** EPSG input projection number */
-        projection: EpsgCode;
-
-        options: {
-            maxConcurrency: number;
+export interface CogProperties {
+    'proj:epsg': EpsgCode;
+    /** Ground Sampling Distance in meters per pixel of the generated imagery */
+    'eo:gsd': number;
+    'linz:basemaps': CogBuild;
+
+    /** How and when this job file was generated */
+}
+
+export interface StacLink {
+    rel: string;
+    href: string;
+    type: string;
+}
+
+export interface StacProvider {
+    name: string;
+    roles: string[];
+    url: string;
+}
+
+export const CogStacVersion = '1.0.0';
+export const CogStacLicense = 'CC-BY-4.0';
+export const CogStacKeywords = ['Imagery', 'New Zealand'];
+export const CogStacExtensions = ['eo', 'proj', 'linz'];
+
+export interface StacCollection<P = Record<string, any>> {
+    /** Unique processing Id */
+    id: string;
+
+    stac_version: string;
+
+    title: string;
+    description?: string;
+
+    license: string;
+
+    properties: P;
```

Comment:
The actually is no properties field in a Stac Collection but this is what LDS are using. There is a summaries field but its format is a bit cumbersome.